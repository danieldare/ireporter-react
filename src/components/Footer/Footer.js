import React from 'react';

const Footer = () => <div>My Footer</div>;

export default Footer;
