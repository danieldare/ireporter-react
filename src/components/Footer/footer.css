/* FOOTER */
.footer_content {
  background: #3c4859;
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding: 50px;
}

.footer-content {
  background: #3c4859;
  text-align: center;
  color: #fff;
  padding: 5px 0;
  align-items: stretch;
  letter-spacing: 0.1rem;
  font-weight: 100;
  font-size: 1.5rem;
}

.footer-content a {
  text-decoration: none;
  color: #fff;
  padding: 0 10px;
  font-weight: 600;
}

.brand-name {
  font-size: 4rem;
  margin: 5px 0;
  color: #fff;
  font-weight: 600;
}

.footer__content__left-brand {
  text-align: center;
}

.brand-name__brief {
  color: #da8367;
  letter-spacing: 0.2rem;
  font-size: 1.5rem;
  background-color: #da8367;
  color: #fff;
  font-weight: 500;
  padding: 7px 15px;
  border-radius: 10px;
}

.footer-bottom {
  background: #4b5461;
  text-align: center;
  padding: 5px;
  font-size: 1.4rem;
  color: #fff;
}
