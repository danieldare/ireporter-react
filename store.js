import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import rootReducer from './src/reducers';

const initialState = {};

const middlewares = [thunk];

if (process.env.NODE_ENV === `development`) {
  middlewares.push(createLogger());
}

const store = createStore(
  rootReducer,
  initialState,
  composeWithDevTools(applyMiddleware(...middlewares))
);

export default store;
