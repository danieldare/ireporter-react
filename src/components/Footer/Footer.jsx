import React from 'react';
import './footer.css';

const Footer = () => (
  <footer className="footer">
    <div className="footer_content">
      <div className="footer__content__left-brand">
        <div className="brand-name">iReporter</div>
        <p className="brand-name__brief">Let's build a better Nation.</p>
      </div>

      <div className="footer-content">
        <a href="##">Terms and Conditions</a>
        <a href="##">| Contact Us</a>
        <a href="##">| Career </a>
      </div>
    </div>
    <div className="footer-bottom">copyright Â© 2018. All rights reserved.</div>
  </footer>
);

export default Footer;
