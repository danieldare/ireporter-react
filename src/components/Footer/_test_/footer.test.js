import React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Provider } from 'react-redux';
import Footer from '../Footer';
import store from '../../../../store';

describe('<Footer />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(
      <Provider store={store}>
        <Footer />
      </Provider>
    );
  });

  it('should match snapshot', () => {
    const tree = toJSON(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
