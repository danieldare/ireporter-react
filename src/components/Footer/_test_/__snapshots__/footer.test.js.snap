// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Footer>
    <footer
      className="footer"
    >
      <div
        className="footer_content"
      >
        <div
          className="footer__content__left-brand"
        >
          <div
            className="brand-name"
          >
            iReporter
          </div>
          <p
            className="brand-name__brief"
          >
            Let's build a better Nation.
          </p>
        </div>
        <div
          className="footer-content"
        >
          <a
            href="##"
          >
            Terms and Conditions
          </a>
          <a
            href="##"
          >
            | Contact Us
          </a>
          <a
            href="##"
          >
            | Career 
          </a>
        </div>
      </div>
      <div
        className="footer-bottom"
      >
        copyright Â© 2018. All rights reserved.
      </div>
    </footer>
  </Footer>
</Provider>
`;
